tests/test_xet_download.py

```diff
@@ -0,0 +1,279 @@
+import os
+from contextlib import contextmanager
+from pathlib import Path
+from unittest.mock import DEFAULT, Mock, patch
+
+from huggingface_hub import snapshot_download
+from huggingface_hub.file_download import (
+    HfFileMetadata,
+    get_hf_file_metadata,
+    hf_hub_download,
+    hf_hub_url,
+    try_to_load_from_cache,
+    xet_get,
+)
+from huggingface_hub.utils import XetMetadata
+
+from .testing_utils import (
+    DUMMY_XET_FILE,
+    DUMMY_XET_MODEL_ID,
+    requires,
+    with_production_testing,
+)
+
+
+@requires("hf_xet")
+@with_production_testing
+class TestXetFileDownload:
+    @contextmanager
+    def _patch_xet_file_metadata(self, with_xet_metadata=True):
```

Comment:
```suggestion
    def _patch_xet_file_metadata(self, with_xet_metadata: bool):
```

(making it required for readability in the tests + it's already used like this)